servant-client/src/Servant/Common/Req.hs

```diff
@@ -214,6 +215,10 @@ instance MonadBaseControl IO ClientM where
   -- restoreM :: StM ClientM a -> ClientM a
   restoreM st = ClientM (restoreM st)
 
+-- | Try clients in order, last error is preserved.
+instance Semigroup (ClientM a) where
+  a <> b = a `catchError` \_ -> b
```

Comment:
I actually don't have strong opinion on this.
- I'd prefer to add `semigroupoids` dependency (it's in `lens` closure), and not define `Semigroup` instance at all, because people have strong opinions for both
    - `(<>) = (<|>)`
    -  `(<>) = liftA2 (<>)`